### PR TITLE
feat: migrate results to vue 3 syntax

### PR DESCRIPTION
**Pull request by lauramargar**
EMP-427

### CODE REVIEW

**Review by CachedaCodes — CHANGES_REQUESTED** [review]

**Comment by support-empathy** [comment]
This is a link to Empathy Platform Docs Portal preview in a development environment. 
 ⚠️ Only for Empathy members. We’re sorry for the inconvenience!
 [Check out EPDocs preview](https://docs-dev.empathy.co/preview/x-1145/index.html)

**Comment by support-empathy** [comment]
This is a link to Empathy Platform Docs Portal preview in a development environment. 
 ⚠️ Only for Empathy members. We’re sorry for the inconvenience!
 [Check out EPDocs preview](https://docs-dev.empathy.co/preview/x-1145/index.html)

File: packages/x-components/src/components/result/base-result-rating.vue [hunk 1]
```diff
@@ -2,6 +2,7 @@
   <component
     :is="link ? 'a' : 'div'"
     v-if="result.rating && result.rating.value"
+    ref="root"
```
Comment (CachedaCodes) [inline packages/x-components/src/components/result/base-result-rating.vue:5]: For consistency, keep the same name for the refs of the root element. Here we use `root` and in other component `el`, should be the same in both (I prefer `el` but it's your call).

**Review by CachedaCodes — CHANGES_REQUESTED** [review]

**Review by CachedaCodes — APPROVED** [review]

File: packages/x-components/src/components/result/base-result-image.vue [hunk 2]
```diff
@@ -41,170 +41,199 @@
...
-     *
-     * @returns True if it should try to load the next image.
-     *
-     * @internal
-     */
-    protected get shouldLoadNextImage(): boolean {
-      const numImagesToLoad = this.showNextImageOnHover && this.userHasHoveredImage ? 2 : 1;
-      return !!this.pendingImages.length && this.loadedImages.length < numImagesToLoad;
-    }
+      const resultImages = ref(props.result.images!);
```
Comment (CachedaCodes) [inline packages/x-components/src/components/result/base-result-image.vue:146]: This breaks the reactivity of the images, use this instead:

```suggestion
      const resultImages = toRef(props.result, 'images');
```

**Review by CachedaCodes — CHANGES_REQUESTED** [review]

**Review by CachedaCodes — APPROVED** [review]

**Review by diegopf — APPROVED** [review]

**Pull request merged**